README.md

```diff
@@ -50,3 +50,86 @@ To run over all the files in the repo manually:
+- `-u hardware_interface:*` will trace all events from the `hardware_interface` namespace.
+- `--list` will list all the events that can be traced.
+
+After that, run the ROS 2 control nodes as usual.
+
+## Inspecting the trace data
+
+To inspect the data of the tracing session, we recommend using `babeltrace2`.
+If created you traced the example above, you can inspect the data with:
+```bash
+babeltrace2 ~/.ros/tracing/abb-hardware-interface | less
+```
+
+This will open a pager with the trace data. It will look something like this:
+
+```
+[14:51:33.967204443] (+0.000080284) sunrise-rtc hardware_interface:write_start: { cpu_id = 0 }, { vpid = 1519705, procname = "ros2_control_no", vtid = 1519779 }, { if_name = "ABBMultiInterfaceHardwareSmall" }
+[14:51:33.967218440] (+0.000013997) sunrise-rtc hardware_interface:write_end: { cpu_id = 0 }, { vpid = 1519705, procname = "ros2_control_no", vtid = 1519779 }, { if_name = "ABBMultiInterfaceHardwareSmall" }
+[14:51:33.967219471] (+0.000001031) sunrise-rtc hardware_interface:write_start: { cpu_id = 0 }, { vpid = 1519705, procname = "ros2_control_no", vtid = 1519779 }, { if_name = "ABBMultiInterfaceHardwareBig" }
+[14:51:33.967221196] (+0.000001725) sunrise-rtc hardware_interface:write_end: { cpu_id = 0 }, { vpid = 1519705, procname = "ros2_control_no", vtid = 1519779 }, { if_name = "ABBMultiInterfaceHardwareBig" }
+[14:51:33.971103015] (+0.003881819) sunrise-rtc hardware_interface:read_start: { cpu_id = 0 }, { vpid = 1519705, procname = "ros2_control_no", vtid = 1519779 }, { if_name = "ABBMultiInterfaceHardwareSmall" }
+[14:51:33.971147404] (+0.000044389) sunrise-rtc hardware_interface:read_end: { cpu_id = 0 }, { vpid = 1519705, procname = "ros2_control_no", vtid = 1519779 }, { if_name = "ABBMultiInterfaceHardwareSmall" }
+[14:51:33.971148861] (+0.000001457) sunrise-rtc hardware_interface:read_start: { cpu_id = 0 }, { vpid = 1519705, procname = "ros2_control_no", vtid = 1519779 }, { if_name = "ABBMultiInterfaceHardwareBig" }
+[14:51:33.971154327] (+0.000005466) sunrise-rtc hardware_interface:read_end: { cpu_id = 0 }, { vpid = 1519705, procname = "ros2_control_no", vtid = 1519779 }, { if_name = "ABBMultiInterfaceHardwareBig" }
+```
+
+In this example we have two controllers, `ABBMultiInterfaceHardwareSmall` and `ABBMultiInterfaceHardwareBig`.
+The events `write_start`, `write_end`, `read_start`, and `read_end` events are logged for both controllers.
+
+# Little bit of help with setting up tracing dependencies
+
+First and foremost, I hope you will not need to do this.
+
+## Tracing with ros2_tracing
+
+The [How to use ros2_tracing to trace and analyze an application](https://docs.ros.org/en/humble/Tutorials/Advanced/ROS2-Tracing-Trace-and-Analyze.html) guide is a good starting point to get tracing going. We are using Humble, which means you will need to build ROS 2 from source. This has also some other implications.
+For example ...
```

Comment:
If the following sections are to be the examples that follow, they should be a different level, so `###`, or bullet points.

But it doesn't sound like that's the case... did you forget the examples?

Reply:
Yes, thank you - the two subsections should be a higher level of heading - `###`.